sql/logictest: TestLogic failed

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2630851&tab=buildLog) on [master@6b58038d1c9848a224b5a9e84f842246fa31aef7](https://github.com/cockroachdb/cockroach/commits/6b58038d1c9848a224b5a9e84f842246fa31aef7):

```
=== RUN   TestLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic149574701
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3316: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic149574701
--- FAIL: TestLogic (802.45s)
=== RUN   TestLogic/local/set
=== PAUSE TestLogic/local/set
=== CONT  TestLogic/local/set
=== CONT  TestLogic/local/set
    logic.go:2265: 
         pq: query execution canceled due to statement timeout
--- done: testdata/logic_test/set with config local: 75 tests, 1 failures
E210131 23:16:21.631414 2577 sql/logictest/logic.go:3548  [-] 44  testdata/logic_test/set:290: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/logic_test/set:290: error while processing
    logic.go:3091: testdata/logic_test/set:290: too many errors encountered, skipping the rest of the input
        --- FAIL: TestLogic/local/set (1.40s)
=== RUN   TestLogic/local
    --- FAIL: TestLogic/local (0.87s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #58269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2633993&tab=buildLog) on [master@98674ef9db7d343f468ff90ee866bb410186acc6](https://github.com/cockroachdb/cockroach/commits/98674ef9db7d343f468ff90ee866bb410186acc6):

```
=== RUN   TestLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic249237256
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3316: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic249237256
--- FAIL: TestLogic (634.43s)
=== RUN   TestLogic/fakedist-disk/lookup_join
=== PAUSE TestLogic/fakedist-disk/lookup_join
=== CONT  TestLogic/fakedist-disk/lookup_join
=== CONT  TestLogic/fakedist-disk/lookup_join
    logic.go:2265: 
         
        testdata/logic_test/lookup_join:609: SELECT * FROM (VALUES (1), (2)) AS u(y) WHERE NOT EXISTS (
          SELECT * FROM t59615 t WHERE u.y = t.y
        )
        expected:
            1
            2
        but found (query options: "") :
            2
            1
--- done: testdata/logic_test/lookup_join with config fakedist-disk: 85 tests, 1 failures
E210201 19:14:10.908152 8017806 jobs/adopt.go:241  [n1] 763  job 629592721765728257: adoption completed with error validation of CHECK "a < 16:::INT8" failed on row: a=17, rowid=629592721757667329
--- total progress: 111755 statements/queries
        --- FAIL: TestLogic/fakedist-disk/lookup_join (3.11s)
=== RUN   TestLogic/fakedist-disk
    --- FAIL: TestLogic/fakedist-disk (0.59s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #58269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2634262&tab=buildLog) on [master@90aad1a266913309a8c54e715316999b84245c49](https://github.com/cockroachdb/cockroach/commits/90aad1a266913309a8c54e715316999b84245c49):

```
=== RUN   TestLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic238307671
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3316: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic238307671
--- FAIL: TestLogic (622.49s)
=== RUN   TestLogic/fakedist-disk/lookup_join
=== PAUSE TestLogic/fakedist-disk/lookup_join
=== CONT  TestLogic/fakedist-disk/lookup_join
=== CONT  TestLogic/fakedist-disk/lookup_join
    logic.go:2265: 
         
        testdata/logic_test/lookup_join:609: SELECT * FROM (VALUES (1), (2)) AS u(y) WHERE NOT EXISTS (
          SELECT * FROM t59615 t WHERE u.y = t.y
        )
        expected:
            1
            2
        but found (query options: "") :
            2
            1
--- done: testdata/logic_test/lookup_join with config fakedist-disk: 85 tests, 1 failures
        --- FAIL: TestLogic/fakedist-disk/lookup_join (3.75s)
=== RUN   TestLogic/fakedist-disk
    --- FAIL: TestLogic/fakedist-disk (0.63s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #58269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2634479&tab=buildLog) on [master@995d7166bf54752ffbc62e0e5e70356cdb1c0db4](https://github.com/cockroachdb/cockroach/commits/995d7166bf54752ffbc62e0e5e70356cdb1c0db4):

```
=== RUN   TestLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic202237676
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3316: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic202237676
--- FAIL: TestLogic (841.48s)
=== RUN   TestLogic/fakedist-disk/lookup_join
=== PAUSE TestLogic/fakedist-disk/lookup_join
=== CONT  TestLogic/fakedist-disk/lookup_join
E210201 21:55:14.169145 8407499 jobs/adopt.go:241  [n1] 714  job 629624386296020993: adoption completed with error validation of NOT NULL constraint failed: validation of CHECK "c IS NOT NULL" failed on row: a=1, rowid=629624385825964033, b=1, c=NULL
=== CONT  TestLogic/fakedist-disk/lookup_join
    logic.go:2265: 
         
        testdata/logic_test/lookup_join:603: SELECT * FROM (VALUES (1), (2)) AS u(y) LEFT JOIN t59615 t ON u.y = t.y
        expected:
            1  NULL  NULL  NULL
            2  NULL  NULL  NULL
            
        but found (query options: "") :
            2  NULL  NULL  NULL
            1  NULL  NULL  NULL
--- done: testdata/logic_test/lookup_join with config fakedist-disk: 84 tests, 1 failures
E210201 21:55:26.886509 7784468 sql/logictest/logic.go:3548  [-] 747  testdata/logic_test/lookup_join:609: too many errors encountered, skipping the rest of the input
    logic.go:3091: 
        testdata/logic_test/lookup_join:609: error while processing
    logic.go:3091: testdata/logic_test/lookup_join:609: too many errors encountered, skipping the rest of the input
        --- FAIL: TestLogic/fakedist-disk/lookup_join (12.81s)
=== RUN   TestLogic/fakedist-disk
    --- FAIL: TestLogic/fakedist-disk (0.54s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #58269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/inverted_join_multi_column
+++ b/pkg/sql/logictest/testdata/logic_test/inverted_join_multi_column
@@ -505,13 +505,13 @@ CREATE TABLE t59615_inv (
   INVERTED INDEX (x, y)
 )
 
-query TITI
+query TITI rowsort
 SELECT * FROM (VALUES ('"a"'::jsonb), ('"b"'::jsonb)) AS u(y) LEFT JOIN t59615_inv t ON t.y @> u.y
 ----
 "a"  NULL  NULL  NULL
 "b"  NULL  NULL  NULL
 
-query T
+query T rowsort
 SELECT * FROM (VALUES ('"a"'::jsonb), ('"b"'::jsonb)) AS u(y) WHERE NOT EXISTS (
   SELECT * FROM t59615_inv t WHERE t.y @> u.y
 )

--- a/pkg/sql/logictest/testdata/logic_test/lookup_join
+++ b/pkg/sql/logictest/testdata/logic_test/lookup_join
@@ -600,13 +600,13 @@ CREATE TABLE t59615 (
 )
 
 # We cannot create a lookup join in this case.
-query IIII
+query IIII rowsort
 SELECT * FROM (VALUES (1), (2)) AS u(y) LEFT JOIN t59615 t ON u.y = t.y
 ----
 1  NULL  NULL  NULL
 2  NULL  NULL  NULL
 
-query I
+query I rowsort
 SELECT * FROM (VALUES (1), (2)) AS u(y) WHERE NOT EXISTS (
   SELECT * FROM t59615 t WHERE u.y = t.y
 )

